ReactSkia/sdk/OnScreenKeyBoard.cpp

```diff
@@ -0,0 +1,618 @@
+    createWindow(screenSize_,createWindowCB);
+
+}
+
+bool OnScreenKeyboard::drawOSK() {
+
+if((oskState_!= OSK_STATE_ACTIVE)) return false;
+
+/*1. Set up OSK Window background*/
+    oskCanvas->clear(bgColor_);
+
+/*2. Draw PlaceHolder*/
+    SkPaint paint;
+    if(oskConfig_.placeHolderName) {
+        SkFont font;
+        paint.setColor(fontColor_);
+        font.setSize(OSK_FONT_SIZE);
+        oskCanvas->drawSimpleText(oskConfig_.placeHolderName,strlen(oskConfig_.placeHolderName), SkTextEncoding::kUTF8,
+                                   oskLayout_.horizontalStartOffset,
+                                   screenSize_.height()*OSK_PH_NAME_VERTICAL_OFFSET,
+                                   font, paint);
+    }
+    paint.setColor((oskConfig_.theme == OSK_LIGHT_THEME) ? OSK_LIGHT_THEME_PH_COLOR: OSK_DARK_THEME_PH_COLOR);
+    SkRect rect=SkRect::MakeXYWH( oskLayout_.horizontalStartOffset,
+                                  screenSize_.height()*OSK_PLACEHOLDER_VERTICAL_OFFSET,
+                                  screenSize_.width()*OSK_PLACEHOLDER_LENGTH,
+                                  OSK_PLACEHOLDER_HEIGHT
+                                );
+    oskCanvas->drawRect(rect,paint);
+
+/*3. Draw PLaceHolder Display String */
+    RNS_LOG_TODO("Update PlaceHolder String on start Up");
+
+/*4. Draw  KeyBoard Layout*/
+    drawKBLayout(oskConfig_.type);
+
+/*5. Finally Listen for  Key Pressed event */
+    std::function<void(rnsKey, rnsKeyAction)> handler = std::bind(&OnScreenKeyboard::onHWkeyHandler,this,
+                                                                    std::placeholders::_1,
+                                                                    std::placeholders::_2);
+    oskEventId_ = NotificationCenter::OSKCenter().addListener("onHWKeyEvent", handler);
+    return true;
+}
+
+bool OnScreenKeyboard::drawKBLayout(OSKTypes oskType) {
+    bool retrunCode{false};
+
+    if((oskState_!= OSK_STATE_ACTIVE)) return false;
+
+    RNS_PROFILE_START(OSKLayoutCreate)
+    createOSKLayout(oskType);
+    RNS_PROFILE_START(OSKDraw)
+//clear KeyBoard Area
+    SkRect kbArea=SkRect::MakeXYWH( oskLayout_.horizontalStartOffset,
+                                    screenSize_.height()*OSK_KB_VERTICAL_OFFSET,
+                                    screenSize_.width(),
+                                    screenSize_.height()- screenSize_.height()*OSK_KB_VERTICAL_OFFSET
+                                  );
+    SkPaint paintObj;
+    paintObj.setColor(bgColor_);
+    oskCanvas->drawRect(kbArea,paintObj);
+    if(oskLayout_.keyInfo && oskLayout_.keyPos) {
+        /*1. Draw Keys */
+        for (unsigned int rowIndex = 0; rowIndex < oskLayout_.keyInfo->size(); rowIndex++) {
+            for (unsigned int keyIndex = 0; keyIndex < oskLayout_.keyInfo->at(rowIndex).size(); keyIndex++) {
+                retrunCode=drawKBKeyFont({keyIndex,rowIndex},fontColor_);
```

Comment:
if drawKBKeyFont returns false , should we return from this function call immediately ?